src/Fail.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace PhpWorkshop\PhpWorkshop;
+
+use PhpWorkshop\PhpWorkshop\Exercise\ExerciseInterface;
+
+/**
+ * Class Fail
+ * @package PhpWorkshop\PhpWorkshop
+ * @author Aydin Hassan <aydin@hotmail.co.uk>
+ */
+class Fail
```

Comment:
yeah not really sure what the best approach on that would be... like you say `Success` doesn't need a reason. 

I'd be inclined to have  a result interface and maybe a result abstract too which would return an empty string on `getReason` ... not 100% though